src/main/java/jnr/unixsocket/UnixSocket.java

```diff
@@ -14,45 +14,147 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ * 
+ * ported from  https://github.com/softprops/unisockets/blob/master/unisockets-core/src/main/scala/Socket.scala 
```

Comment:
Ports are considered derivative works, so I'd like to hear from @softprops that it's ok to include this code under jnr's Apache-2 license.